src/neo/SmartContract/Native/Votes/VoteContract.cs

```diff
@@ -0,0 +1,428 @@
+                IsSequence = false
+            };
+            if (RegisterVote(engine.Snapshot, createState))
+            {
+                return tx.Hash.ToArray();
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        [ContractMethod(0_01000000, ContractParameterType.ByteArray)]
+        private StackItem MultiVote(ApplicationEngine engine, VMArray args)
+        {
+            if (args[0] is null || args[1] is null || args[2] is null) return false;
+            UInt256 TxHash = new UInt256(args[0].GetSpan());
+            var id = Crypto.Hash160(VoteCreateState.ConcatByte(TxHash.ToArray(), (engine.CallingScriptHash ?? this.Hash).ToArray()));
+            UInt160 voter = new UInt160(args[1].GetSpan());
+            if (!InteropService.Runtime.CheckWitnessInternal(engine, voter)) return false;
+            StorageKey AccessKey = CreateStorageKey(Prefix_AccessControl, id);
+            StorageItem access_state = engine.Snapshot.Storages.TryGet(AccessKey);
+            if (!(access_state is null))
+            {
+                HashSet<UInt160> accessVoter = ConvertBytesToUserArray(access_state.Value);
+                if (!accessVoter.Contains(voter))
+                {
+                    return false;
+                }
+            }
+            StorageKey InfoKey = CreateStorageKey(Prefix_CreateVote, id);
+            StorageItem info_state = engine.Snapshot.Storages.TryGet(InfoKey);
+            MultiCandidate candidate = Neo.IO.Helper.AsSerializable<MultiCandidate>(args[2].GetSpan());
+            VoteState voteState = new VoteState(voter, candidate);
+            return (AddVote(engine.Snapshot, voteState, id));
+        }
+
+        [ContractMethod(0_01000000, ContractParameterType.ByteArray)]
+        private StackItem SingleVote(ApplicationEngine engine, VMArray args)
+        {
+            if (args[0] is null || args[1] is null || args[2] is null) return false;
+            UInt256 TxHash = new UInt256(args[0].GetSpan());
+            var id = Crypto.Hash160(VoteCreateState.ConcatByte(TxHash.ToArray(), (engine.CallingScriptHash ?? this.Hash).ToArray()));
+            UInt160 voter = new UInt160(args[1].GetSpan());
+            if (!InteropService.Runtime.CheckWitnessInternal(engine, voter)) return false;
+            StorageKey AccessKey = CreateStorageKey(Prefix_AccessControl, id);
+            StorageItem access_state = engine.Snapshot.Storages.TryGet(AccessKey);
+            if (!(access_state is null))
+            {
+                HashSet<UInt160> accessVoter = ConvertBytesToUserArray(access_state.Value);
+                if (!accessVoter.Contains(voter))
+                {
+                    return false;
+                }
+            }
+            StorageKey InfoKey = CreateStorageKey(Prefix_CreateVote, id);
+            StorageItem info_state = engine.Snapshot.Storages.TryGet(InfoKey);
```

Comment:
Could we unify the naming? like camel-case?

Reply:
Get it,  will solve them soon.